orchagent/mirrororch.cpp

```diff
@@ -0,0 +1,745 @@
+    }
+
+    m_syncdMirrors.erase(sessionIter);
+}
+
+bool MirrorOrch::setSessionState(const string& name, MirrorEntry& session)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_INFO("Setting mirroring sessions %s state\n", name.c_str());
+
+    vector<FieldValueTuple> fvVector;
+
+    string status = session.status ? MIRROR_SESSION_STATUS_ACTIVE : MIRROR_SESSION_STATUS_INACTIVE;
+
+    FieldValueTuple t(MIRROR_SESSION_STATUS, status);
+    fvVector.push_back(t);
+
+    m_mirrorTableProducer.set(name, fvVector);
+
+    return true;
+}
+
+bool MirrorOrch::getNeighborInfo(const string& name, MirrorEntry& session)
+{
+    SWSS_LOG_ENTER();
+
+    NeighborEntry neighbor;
+    MacAddress mac;
+
+    assert(session.nexthopInfo.resolved);
+
+    SWSS_LOG_INFO("Getting neighbor info for %s session\n", name.c_str());
+
+    if (!m_neighOrch->getNeighborEntry(session.nexthopInfo.nexthop, neighbor, mac))
+    {
+        session.neighborInfo.resolved = false;
+        return false;
+    }
+
+    return getNeighborInfo(name, session, neighbor, mac);
+}
+
+bool MirrorOrch::getNeighborInfo(const string& name, MirrorEntry& session, const NeighborEntry& neighbor, const MacAddress& mac)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_INFO("Getting neighbor info for %s session\n", name.c_str());
+
+    session.neighborInfo.resolved = false;
+    session.neighborInfo.neighbor = neighbor;
+    session.neighborInfo.mac = mac;
+
+    // Get port operation should not fail;
+    if (!m_portsOrch->getPort(session.neighborInfo.neighbor.alias, session.neighborInfo.port))
+    {
+        SWSS_LOG_ERROR("Failed to get port for %s alias\n", session.neighborInfo.neighbor.alias.c_str());
+        assert(false);
+    }
+
+    if (session.neighborInfo.port.m_type == Port::VLAN)
+    {
+        session.neighborInfo.vlanId = session.neighborInfo.port.m_vlan_id;
+
+        Port member;
+        if (!m_fdbOrch->getPort(session.neighborInfo.mac, session.neighborInfo.vlanId, member))
+        {
+            return false;
+        }
+
+        session.neighborInfo.portId = member.m_port_id;
+        session.neighborInfo.resolved = true;
+
+        return true;
+    }
+    else if (session.neighborInfo.port.m_type == Port::LAG)
+    {
+        session.neighborInfo.vlanId = session.neighborInfo.port.m_port_vlan_id;
+
+        if (session.neighborInfo.port.m_members.empty())
+        {
+            return false;
+        }
+
+        Port lagMember;
+        if (!m_portsOrch->getPort(*session.neighborInfo.port.m_members.begin(), lagMember))
+        {
+            SWSS_LOG_ERROR("Failed to get port for %s alias\n", session.neighborInfo.port.m_members.begin()->c_str());
+            assert(false);
+        }
+
+        session.neighborInfo.portId = lagMember.m_port_id;
+        session.neighborInfo.resolved = true;
+    }
+    else
+    {
+        session.neighborInfo.portId = session.neighborInfo.port.m_port_id;
+        session.neighborInfo.vlanId = session.neighborInfo.port.m_port_vlan_id;
```

Comment:
>m_port_vlan_id [](start = 64, length = 14)

when the port is physical port or lag, the m_port_vlan_id is probably 0, will this work?